### DISCUSSION TITLE
Deprecation Warnings

### DISCUSSION THREAD

**Discussion by tobfd** [Pycord-Development/pycord#2468] [Q&A] [answered]
how can i prevent that when i run my py-cord bot and also when i press a button there are different “DeprecationWarnings” when i start the bot there are 2 but when i click on a button there are about 20. i use the @feat_ua branch.
![image](https://github.com/Pycord-Development/pycord/assets/77446267/72135794-263c-4eb1-aedb-8519625d6399)

**Comment by tobfd** [comment]
![image](https://github.com/Pycord-Development/pycord/assets/77446267/e1b35c98-ccb1-492d-b231-0c881c91f7c4)

**Comment by NeloBlivion** [comment, marked as answer]
That is the point of the warning, to encourage you to migrate to new functions. User apps deprecate certain features, so you should read the generated docs and update your code accordingly (see https://pycord--2409.org.readthedocs.build/en/2409/api/models.html#discord.Message.interaction)

**Reply by plun1331** [reply]
It should also be noted that your second screenshot is from an external library, and you should ask them to update their library.
